docs/docs/reference/fast-refresh.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Fast Refresh
+---
+
+Fast Refresh is an implementation of Hot Reloading with full support from React and it replaces purely userland solutions like [`react-hot-loader`](https://github.com/gaearon/react-hot-loader). It is a feature that allows you to get direct feedback for changes in your React components: A change in your editor immediately updates the browser view and if you have errors in your code the browser view displays an error overlay. Fast Refresh is enabled by default in Gatsby's development server and should show your changes within a second, without losing component state which results in an overall better development experience (DX).
```

Comment:
One other idea: How hard would it be to make a short video like this one from the [hot reloading React docs](https://reactnative.dev/blog/2016/03/24/introducing-hot-reloading), showing the difference in DX for when fast refresh is enabled vs. disabled?

Seeing the difference was helpful for me when I was figuring out what fast refresh actually does. Truly just a suggestion though.

Reply:
I'd file this under nice-to-have for the future :D